src/connection.ts

```diff
@@ -173,6 +173,9 @@ export function registerLanguageHandler(messageEmitter: MessageEmitter, messageW
 			case 'exit':
 				// Ignore exit notification, it's not the responsibility of the TypeScriptService to handle it,
 				// but the TCP / STDIO server which needs to close the socket or kill the process
+				for (const subscription of subscriptions.values()) {
```

Comment:
Good question. This code is for side effects. `iterate`/Observables are great where you have a return value and need to filter/map/etc. But `forEach` has many disadvantages:
- you can't step through the loop with a debugger
- no `break` keyword
- TS compiler can't narrow types of properties because the callback could be called after the object was mutated

so in general, prefer `map`/`filter` over `for..of`, but `for..of` over `forEach`.